library/Yesod/Session/Persist.hs

```diff
@@ -0,0 +1,80 @@
+module Yesod.Session.Persist
+  ( -- * Setup
+    makeSessionBackend
+  , SessionConfiguration (..)
+
+    -- * Options
+  , Options (..)
+  , defaultOptions
+  , hoistOptions
+
+    -- * Timing
+  , TimingOptions (..)
+  , defaultTimingOptions
+
+    -- * Timeout
+  , Timeout (..)
+  , defaultTimeout
+
+    -- * Transport security
+  , TransportSecurity (..)
+
+    -- * Session data model
+  , Session (..)
+  , SessionKey (..)
+  , Time (..)
+
+    -- * Randomization
+  , Randomization (..)
+  , defaultRandomization
+  , deterministicallyRandom
+  , DeterministicRandomization (..)
+
+    -- * Storage
+  , SessionPersistence (..)
+  , StorageException (..)
+
+    -- * Key rotation
+  , rotateSessionKey
+  , assignSessionKeyRotation
+  , KeyRotation (..)
+
+    -- * Freezing
+  , disableSessionManagement
+  , assignSessionFreeze
+  , SessionFreeze (..)
+
+    -- * Session map embedding
+  , SessionEmbeddings (..)
+  , HasSessionEmbeddings (..)
+  , Embedding (..)
+  , SessionMapEmbedding
+  , MapOperations (..)
+  , bsKeyEmbedding
+  , dimapEmbedding
+  , showReadKeyEmbedding
+
+    -- * Comparison
+  , Comparison (..)
+  , differsOn
```

Comment:
Is all of this needed for the common case?

This is a complex beast we have here, necessarily so, but I'm wondering if the top-level module exposing only what's needed for a simple or common use case (e.g. `makeSessionBackend`, `defaultOptions`, and the types involved), then having to reach for other modules to do more things (such as change the randomization) could actually be a feature, a useful discoverability point, so you can read and ingest the haddocks in chucks as you need different things, instead of being met with this whole interface up front.

Sorry to make another "hey, why not re-org all your modules" suggestion again though.

Reply:
Much is needed upfront, but options that aren't really expected to be changed (randomization and map embeddings) could be shoved off onto other modules. I didn't feel like there was a discoverability issue because the most important things are at the top, and in the generated haddock you can then click through the types of the fields for `SessionConfiguration`, `Options`, etc. to be taken to anything you care to explore further down the page.

Reply:
Wouldn't be too much of a reorganization, because it would only amount to reorganizing the user-facing facade.